src/main/pages/che-7/end-user-guide/ref_devfile-reference.adoc

```diff
@@ -1,23 +1,251 @@
 [id="devfile-reference_{context}"]
 = Devfile reference
 
-This paragraph is the reference module introduction and is only optional. Include it to provide a short overview of the module.
+This section contains devfile reference and instructions on how to use the various elements that devfiles consist of.
 
-A reference module provide data that users might want to look up, but do not need to remember. It has a very strict structure, often in the form of a list or a table. A well-organized reference module enables users to scan it quickly to find the details they want. AsciiDoc markup to consider for reference data:
+== Adding components to a devfile
 
+Each component in a single devfile must have a unique name.
```

Comment:
"name" is misleading here. Each type of component has a different way of identification. plugins have ids, dockerimages have `image` and openshift/kubernetes lists have `reference` used as an identifier. Any two plugins with the same identifier need to be disambiguated by an `alias` which needs to be unique amongst all components in the devfile.

Reply:
OK, so, identifiers can be `id`, `image`, or `reference`. And if any of those coincide, an `alias` is also needed?

Is one alias enough, or do all comps with same identifiers need an alias?

Reply:
1 alias should be enough. Also, an alias is **required** if the component is referenced in some commands.

Reply:
> an alias is required if the component is referenced in some commands

I'm getting a bit lost... Could you please provide a list of all cases in which an alias is required?

Reply:
The sentence `Each component in a single devfile must have a unique name.` was there because in the past each component had a mandatory field `name`. `alias` is required only if you want to reference the component in a command.